plugins/datasetsplit_procedure.cc

```diff
@@ -0,0 +1,246 @@
+
+SplitProcedureConfigDescription::
+SplitProcedureConfigDescription()
+{
+    InputQuery labels;
+    std::vector<PolyConfigT<Dataset>> outputDatasets;
+    std::vector<float> splits;
+
+    addField("labels", &SplitProcedureConfig::labels,
+             "Row of labels to distribute among the splits while keeping their respective proportions");
+    addField("outputDatasets", &SplitProcedureConfig::outputDatasets,
+             "Configuration for output dataset");
+    addField("splits", &SplitProcedureConfig::splits,
+             "Proportion of rows to put into each dataset. The sum of the input should approximate to 1.0");
+    addField("foldImportance", &SplitProcedureConfig::foldImportance,
+             "Importance of respecting the splits versus the distribution of the labels",
+             1.0f);    
+    addParent<ProcedureConfig>();
+}
+
+
+/*****************************************************************************/
+/* DATASET SPLIT PROCEDURE                                                   */
+/*****************************************************************************/
+
+SplitProcedure::
+SplitProcedure(MldbServer * owner,
+            PolyConfig config,
+            const std::function<bool (const Json::Value &)> & onProgress)
+    : Procedure(owner)
+{
+    this->procConfig = config.params.convert<SplitProcedureConfig>();
+}
+
+Any
+SplitProcedure::
+getStatus() const
+{
+    return Any();
+}
+
+RunOutput
+SplitProcedure::
+run(const ProcedureRunConfig & run,
+    const std::function<bool (const Json::Value &)> & onProgress) const
+{
+    auto runProcConf = applyRunConfOverProcConf(procConfig, run);
+    SqlExpressionMldbScope context(server);
+    ConvertProgressToJson convertProgressToJson(onProgress);
+
+    auto boundDataset = runProcConf.labels.stm->from->bind(context, convertProgressToJson);
+
+    std::vector<RowPath> rowPaths;
+
+    //Get all the row names selected by the specified FROM/WHERE
+    //in a deterministic order
+     auto processor = [&] (NamedRowValue & row_,
+                           const std::vector<ExpressionValue> & extraVals)
+        {
+            rowPaths.push_back(std::move(row_.rowName));
+            return true;            
+        };
+    SelectExpression select("1");
+    std::vector<std::shared_ptr<SqlExpression> > extra;
+    BoundSelectQuery(select,
+                     *boundDataset.dataset,
+                     boundDataset.asName, runProcConf.labels.stm->when,
+                     *runProcConf.labels.stm->where,
+                     runProcConf.labels.stm->orderBy, extra)
+        .execute({processor,false/*processInParallel*/},
+                 runProcConf.labels.stm->offset,
+                 runProcConf.labels.stm->limit,
+                 nullptr /* progress */);
+
+    //Shuffle to prevent any aliasing effect
+    std::minstd_rand rng;
+    std::shuffle(rowPaths.begin(), rowPaths.end(), rng);
+
+    size_t numFolds = runProcConf.splits.size();
```

Comment:
As I replied on another comment, those are not actual requirements. I'll add them since you insist.